### PR TITLE
Update kafka channel samples with configmap changes

### PR DESCRIPTION
**Pull request by eric-sap**
This updates the samples/kafka/channel README with information related to the configmap consolidation 

## Proposed Changes

- Describe configmap updates (mainly SASL options and miscellaneous channel configuration) from https://github.com/knative-sandbox/eventing-kafka/pull/622
- Update links and call out areas where the difference between consolidated vs. distributed channel is important

### CODE REVIEW

File: docs/eventing/samples/kafka/channel/README.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ EOF
 ## Specifying the default channel configuration
 
 To configure the usage of the `KafkaChannel` CRD as the
-[default channel configuration](../../../channels/default-channels.md), edit the
+[default channel configuration](../../../channels/channel-types-defaults.md), edit the
```
Comment (abrennan89) [inline docs/eventing/samples/kafka/channel/README.md:40]:
```suggestion
[default channel configuration](../../../channels/channel-types-defaults), edit the
```

File: docs/eventing/samples/kafka/channel/README.md [hunk 2]
```diff
@@ -178,8 +181,8 @@ To use TLS authentication you must create:
 
 **NOTE:** Kafka channels require these files to be in `.pem` format. If your files are in a different format, you must convert them to `.pem`.
 
-
-1. Create the certificate files as secrets in your chosen namespace:
+For the consolidated channel type, create the certificate files as secret
```
Comment (abrennan89) [inline docs/eventing/samples/kafka/channel/README.md:184]: Procedures should remain as numbered steps

File: docs/eventing/samples/kafka/channel/README.md [hunk 3]
```diff
@@ -220,29 +239,87 @@ $ kubectl create secret --namespace <namespace> generic <kafka-auth-secret> \
...
 $ kubectl create secret --namespace <namespace> generic <kafka-auth-secret> \
-    --from-literal=tls.enabled=true \
+  --from-literal=tls.enabled=true \
   --from-literal=password="SecretPassword" \
   --from-literal=saslType="SCRAM-SHA-512" \
   --from-literal=user="my-sasl-user"
 ```
 
-Reference your secret and the namespace of the secret in the `config-kafka` ConfigMap:
+*NOTE:* It is important to use the same keys; `user`, `password` and `saslType`.
```
Comment (abrennan89) [inline docs/eventing/samples/kafka/channel/README.md:253]:
```suggestion
!!! note
    It is important to use the same keys; `user`, `password` and `saslType`.
```
There's new formatting to use admonitions in mkdocs: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage

File: docs/eventing/samples/kafka/channel/README.md [hunk 4]
```diff
@@ -220,29 +239,87 @@ $ kubectl create secret --namespace <namespace> generic <kafka-auth-secret> \
...
+        SASL:
+          Enable: true
+...
 ```
+
+### All authentication methods
+
+After creating the secret for your desired authentication type using the above
+steps, reference that secret and the namespace of the secret in the
+`config-kafka` ConfigMap:
```
Comment (abrennan89) [inline docs/eventing/samples/kafka/channel/README.md:294]:
```suggestion
After you have created the secret for your desired authentication type by using the previous
steps, reference the secret and the namespace of the secret in the `config-kafka` ConfigMap:
```

File: docs/eventing/samples/kafka/channel/README.md [hunk 5]
```diff
@@ -220,29 +239,87 @@ $ kubectl create secret --namespace <namespace> generic <kafka-auth-secret> \
...
+...
+data:
+  eventing-kafka: |
+    kafka:
+      authSecretName: <kafka-auth-secret>
+      authSecretNamespace: <namespace>
+...
+ ```
+
+*NOTE:* The default secret name and namespace are `kafka-cluster` and
```
Comment (abrennan89) [inline docs/eventing/samples/kafka/channel/README.md:305]: Again, admonition formatting please 🙂

**Review by abrennan89 — CHANGES_REQUESTED** [review]
Some suggestions about layout and formatting.
Procedures should have numbered steps. Optional steps can be marked like:
"1. Optional. Whatever the step is.."

File: docs/eventing/samples/kafka/channel/README.md (hunk 1 above)
Comment (eric-sap) [reply docs/eventing/samples/kafka/channel/README.md]: I removed the MD suffix from this link, but now it appears invalid in my editor.  I assume that in final docs, a link to an MD file is valid without the file extension?

File: docs/eventing/samples/kafka/channel/README.md (hunk 2 above)
Comment (eric-sap) [reply docs/eventing/samples/kafka/channel/README.md]: I have replaced the numbering

File: docs/eventing/samples/kafka/channel/README.md (hunk 3 above)
Comment (eric-sap) [reply docs/eventing/samples/kafka/channel/README.md]: I have replaced this and the other collection of old *NOTE:* formatting in the file with the new style.

File: docs/eventing/samples/kafka/channel/README.md (hunk 5 above)
Comment (eric-sap) [reply docs/eventing/samples/kafka/channel/README.md]: as suggested

File: docs/eventing/samples/kafka/channel/README.md (hunk 1 above)
Comment (eric-sap) [reply docs/eventing/samples/kafka/channel/README.md]: Ah I see in the deployed preview that this does in fact work

File: docs/eventing/samples/kafka/channel/README.md (hunk 4 above)
Comment (eric-sap) [reply docs/eventing/samples/kafka/channel/README.md]: Modified the style as suggested

**Comment by eric-sap** [comment]
I believe I have adjusted the spacing and numbering properly now.  I appreciate your patience.

**Comment by abrennan89** [comment]
Thanks @eric-sap - one last thing, in the preview link it looks like some of the numbering / indenting the code blocks for some of the steps is off. You might need to indent the code blocks after the numbered step. Please use the preview to check formatting.

**Comment by eric-sap** [comment]
Okay, I think I understand the indentation and spacing requirements better now.  It is somewhat different than what I remember before, but those are days far gone by.  The preview is extremely convenient.

**Comment by abrennan89** [comment]
Thanks @eric-sap - this is much better, however the admonitions / notes still look like they're breaking the numbering in a couple of places. Also, re `cat <<-EOF | kubectl apply -f -` - please see the draft [style guide](https://docs.google.com/document/d/1tigolRzKhronvgoAJfszE-XHWWd0LQ8o_j0D73O9KvM/edit#heading=h.qqz6in2pxqzk) for the kubectl commands.

@omerbensaadon can you please ensure that the style guide etc is moved to the site or somewhere more accessible for folks to find it; this should be part of the contrib guidelines work we talked about earlier in Slack IMO.

**Comment by eric-sap** [comment]
Unless the numbered steps are supposed to increment regardless of the section, my intent was to restart from 1 whenever I switch from "consolidated" to "distributed" channel type instructions.  If this is confusing I can look into re-arranging that but there is a natural disjointedness to the instructions depending on which channel type the user installed.

**Comment by abrennan89** [comment]
Yeah that's pretty confusing to me tbh 😕 Maybe split this into two different procedures and give them specific headings for each channel type so a user can find the procedure they need more easily?

**Review by abrennan89 — APPROVED** [review]
/lgtm

**Pull request merged**